plugins/modules/site_workflow_manager.py

```diff
@@ -890,28 +1606,128 @@ def delete_single_site(self, site_id, site_name):
             )
 
             if response and isinstance(response, dict):
-                self.log("Received API response from 'delete_site': {0}".format(str(response)), "DEBUG")
+                self.log("Received API response from 'delete_site': {0}".format(
+                    str(response)), "DEBUG")
                 executionid = response.get("executionId")
 
                 while True:
                     execution_details = self.get_execution_details(executionid)
                     if execution_details.get("status") == "SUCCESS":
                         self.status = "success"
                         self.deleted_site_list.append(site_name)
-                        self.log("Site '{0}' deleted successfully".format(site_name), "INFO")
+                        self.log("Site '{0}' deleted successfully".format(
+                            site_name), "INFO")
                         break
                     elif execution_details.get("bapiError"):
-                        self.log("Error response for 'delete_site' execution: {0}".format(execution_details.get("bapiError")), "ERROR")
-                        self.module.fail_json(msg=execution_details.get("bapiError"), response=execution_details)
+                        self.log("Error response for 'delete_site' execution: {0}".format(
+                            execution_details.get("bapiError")), "ERROR")
+                        self.module.fail_json(msg=execution_details.get(
+                            "bapiError"), response=execution_details)
                         break
 
         except Exception as e:
             self.status = "failed"
-            self.msg = "Exception occurred while deleting site '{0}' due to: {1}".format(site_name, str(e))
+            self.msg = "Exception occurred while deleting site '{0}' due to: {1}".format(
+                site_name, str(e))
             self.log(self.msg, "ERROR")
 
         return self
 
+    def delete_floor(self, site_name):
+        """
+        Deletes a floor site by ID.
+
+        Parameters:
+            site_id (str): The ID of the floor site to be deleted.
+            site_name (str): The name of the floor site to be deleted.
+
+        Returns:
+            dict: The API response from the 'deletes_a_floor' operation.
+        """
+        site_id = self.have.get("site_id")
+        try:
+            self.log(
+                "Deleting floor site: {0} with ID: {1}".format(site_name, site_id), "INFO")
+            response = self.dnac._exec(
+                family="site_design",
+                function="deletes_a_floor",
+                op_modifies=True,
+                params={'id': site_id},
+            )
+            return response
+
+        except Exception as e:
+            site_id = self.have.get("site_id")
```

Comment:
Already you fetch the site_id at line no 1647 so no need to repeat the code.

Reply:
Removed